### ISSUE TITLE
Please let tf.keras.metrics.MeanIoU support y_pred from logits

### ISSUE THREAD

**Issue by motionlife** [tensorflow/tensorflow#45598]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
 tf.keras.metrics.MeanIoU doesn't have an option that allow y_pred passed as logits, we want to pass y_pred as a logits tensor, you can simply add an optional step into the ```update_state``` function of this class: 
```
    y_pred = math_ops.argmax(y_pred, axis=-1)
```
**Will this change the current api? How?**
Yes, add one parameter from_logits, you can keep default as False though:
```tf.keras.metrics.MeanIoU(num_classes, from_logits=False, name=None, dtype=None)```

**Who will benefit with this feature?**
Computer vision object detection/segmentation community

**Any Other info.**

**Comment by bhack** [comment]
@motionlife Can you prepare a small PR?

**Comment by motionlife** [comment]
> @motionlife Can you prepare a small PR?

Sure, just did :)

**Comment by fchollet** [comment]
Thanks for the feature request -- moving discussion to the PR you created.
